### PR TITLE
fix(downmerge): merge to the current release branch

### PR DESCRIPTION
**Pull request by abachyla**
We found an issue when we merge a release branch to master, it doesn't do downmerge because there is nothing to commit, but there is an ignored merge commit. When we open a new pr from the same release branch it shows that the release branch is outdated and we have to do downmerges manually: example https://github.com/River-iGaming/azurolongo.jackpotbob/pull/518

Merge to master - Downmerge to release branches and develop
  * develop
  * same `release/1.2.x` version
  * higher `release/1.3.x` version
Screenshot 2023-12-11 at 11 50 02

Downmerge to develop if nothing to commit
Screenshot 2023-12-11 at 12 06 01

### CODE REVIEW

File: .github/workflows/downmerge-releases.yml [hunk 1]
```diff
@@ -44,41 +47,47 @@ jobs:
...
             BRANCH_VERSION=$(echo $branch | cut -d'/' -f3)
             MAJOR_MINOR_BRANCH_VERSION=$(echo $BRANCH_VERSION | cut -d'.' -f1,2)
+            PATCH_BRANCH_VERSION=$(echo $BRANCH_VERSION | cut -d'.' -f3,4)
             MAJOR_MINOR_PACKAGE_VERSION=$(echo $BASE_PACKAGE_VERSION | cut -d'.' -f1,2)
             branch="${branch#origin/}"
+              echo "::notice :: branch: $branch"
+              echo "::notice :: patch version x"
+              echo "::notice :: MAJOR_MINOR_BRANCH_VERSION: $MAJOR_MINOR_BRANCH_VERSION"
+              echo "::notice :: MAJOR_MINOR_PACKAGE_VERSION: $MAJOR_MINOR_PACKAGE_VERSION"
+            if [ "$PATCH_BRANCH_VERSION" == "x" ]; then
```
Comment (abachyla) [inline .github/workflows/downmerge-releases.yml:57]: todo: need to test more thoroughly

File: .github/workflows/downmerge-releases.yml [hunk 2]
```diff
@@ -44,41 +47,48 @@ jobs:
...
+            PATCH_BRANCH_VERSION=$(echo $BRANCH_VERSION | cut -d'.' -f3,4)
             MAJOR_MINOR_PACKAGE_VERSION=$(echo $BASE_PACKAGE_VERSION | cut -d'.' -f1,2)
             branch="${branch#origin/}"
+            echo "::notice :: branch: $branch"
+            echo "::notice :: BRANCH_VERSION: $BRANCH_VERSION"
+            echo "::notice :: patch version $PATCH_BRANCH_VERSION"
+            echo "::notice :: MAJOR_MINOR_BRANCH_VERSION: $MAJOR_MINOR_BRANCH_VERSION"
+            echo "::notice :: MAJOR_MINOR_PACKAGE_VERSION: $MAJOR_MINOR_PACKAGE_VERSION"
+            if [ "$branch" != "$CURRENT_BRANCH" ]; then
+              if [[ $branch == "develop" || $MAJOR_MINOR_BRANCH_VERSION > $MAJOR_MINOR_PACKAGE_VERSION || ("$PATCH_BRANCH_VERSION" == "x" && "$MAJOR_MINOR_BRANCH_VERSION" == "$MAJOR_MINOR_PACKAGE_VERSION") ]]; then
```
Comment (abachyla) [inline .github/workflows/downmerge-releases.yml:59]: considering to remove `"$PATCH_BRANCH_VERSION" == "x"` from the condition, probably won't be needed, because maybe we would like to do merges to branches`release/1.2.3`, not only to `release/1.2.x`

File: .github/workflows/downmerge-releases.yml [hunk 3]
```diff
@@ -23,7 +23,17 @@ jobs:
           fetch-depth: 0
           token: ${{ secrets.github-token }}
 
+      - name: Find Opened Pull Request
+        id: find-pull-request
+        run: |
+          PULL_REQUEST_NUMBER=$(gh search prs $COMMIT_SHA --state=open --json number -L 1 -q ".[].number")
```
Comment (abachyla) [inline .github/workflows/downmerge-releases.yml:29]: find 1st opened pr associated with current commit

**Comment by abachyla** [comment]
no review, found a bug that it still doesn't merge from master to the branch with the same minor version
even it tells that nothing to commit but master has +1 merge commit
Screenshot 2023-12-08 at 17 18 51

File: .github/workflows/downmerge-releases.yml [hunk 4]
```diff
@@ -44,32 +54,34 @@ jobs:
...
-
-              git checkout $branch
+                git checkout $branch
 
-              # Merge changes from the remote branch into branch
-              if git merge --no-commit --no-ff --no-edit ${CURRENT_BRANCH}; then
-                if [ -z "$(git status --porcelain)" ]; then
+                # Merge changes from the remote branch into branch
+                if git merge --no-commit --no-ff --no-edit ${CURRENT_BRANCH}; then
+                if [[ $IS_CURRENT_RELEASE_BRANCH == false && (-z "$(git status --porcelain)") ]]; then
```
Comment (abachyla) [inline .github/workflows/downmerge-releases.yml:69]: case:
we merge a branch `release/z.y.x` to master and we need to do downmerge from master to `release/z.y.x` to have a merge commit even if there are no changes. If we don't do this and create another pr from `release/z.y.x` to `master` we will see a message `Branch is outdated` and will have to do downmerge manually.

**Review by kurtaqui — APPROVED** [review]
